### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by chaimchessrak-vayyar**

### CODE REVIEW

File: scripts/unit_tests.py [hunk 1]
```diff
@@ -0,0 +1,259 @@
+import math
+import os
+import unittest
+from typing import List, Set, Dict
```
Comment (alons-vayyar) [inline scripts/unit_tests.py:4]: needed?

File: scripts/unit_tests.py [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+
```
Comment (alons-vayyar) [inline scripts/unit_tests.py:5]: delete row.

File: scripts/unit_tests.py [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+from scripts import skeleton
+
+
+class UnitTests(unittest.TestCase):
+
+    def setUp(self) -> None:
```
Comment (alons-vayyar) [inline scripts/unit_tests.py:11]: delete None.

File: scripts/unit_tests.py [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+            return wafer_text
+
+        total_dimension = dimension_1 * dimension_2
+        max_representing_number = pow(len(translator), total_dimension)
+        for representing_number in range(max_representing_number):
+            result_wafer_text = make_input_from_number(representing_number)
+            yield result_wafer_text
+
+    def test_on_file(self):
+        path_for_input_file = os.path.join(self.unit_tests_directory, 'unit_test_1_input.txt')
```
Comment (alons-vayyar) [inline scripts/unit_tests.py:163]: change os.path to pathlib.Path , in the hole file.

File: scripts/unit_tests.py [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+        ..
+        XX
+        ...
+        X1X
+        111
+        foooooo!?>>...XX11fooo!
+        foooooo!?>>...XX11fooo!
+        fo!?>>...XX11foo!\n
+        """
+        actual_result = skeleton.clear_from_garbage(garbage_text)
```
Comment (alons-vayyar) [inline scripts/unit_tests.py:219]: change "garbage"

File: scripts/unit_tests.py [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+import unittest
+from typing import List, Set, Dict
+
+from scripts import skeleton
+
+
+class UnitTests(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.unit_tests_directory = '../unit_tests'
```
Comment (alons-vayyar) [inline scripts/unit_tests.py:12]: a much better way:
https://stackoverflow.com/questions/3430372/how-do-i-get-the-full-path-of-the-current-files-directory

File: scripts/unit_tests.py (hunk 2 above)
Comment (chaimchessrak-vayyar) [reply scripts/unit_tests.py:5]: When I do Ctrl alt l this row is the result so I assume 
this is the standard in python.

**Pull request merged**